### DISCUSSION TITLE
How to update a single type from the Strapi API?

### DISCUSSION THREAD

**Discussion by abdonrd** [strapi/strapi#6120] [Legacy Topics]
What is the way to update a single type from the Strapi API?

This doesn't work:

```javascript
await strapi.query('single-type-name').update({
  title: 'New title!'
});
```

And to query?

```javascript
await strapi.query('single-type-name').find();
```

**Comment by ramlez** [comment]
@abdonrd you were very close :) 

Take look at this section: https://strapi.io/documentation/3.0.0-beta.x/concepts/models.html#concept

> They are also available through the `strapi.models` and `strapi.api.**.models` **global variables**. **Usable everywhere in the project**, they **contain the ORM model object** that they refer to. By convention, a model's name should be written in lowercase.

So in other words try to do: 

```js
await strapi.model["single-type-name"].update(
  { _id: 123 },
  {
    $set: {
      title: "New title!",
    },
  }
);

// or
await strapi.model.modelName.findOne()
```

**Comment by lauriejim** [comment]
Hello, the `query` of a Single Type will have the same functions as a Collection Type.
The query system is general for all models.

Base on the source code of the Single Types core controllers/service.

- Controllers:  https://github.com/strapi/strapi/blob/b5ec9cb1c85638f5158ee431e4c7b00d615e5816/packages/strapi/lib/core-api/controller.js#L20
- Services: https://github.com/strapi/strapi/blob/c0bf015eb60ef77a77bca3e2e880a34abba0c187/packages/strapi/lib/core-api/service.js#L19

We can see the service create an abstraction layer to not use the entry id but behind the scene Strapi is doing a find before the update to get the entry id.

**Reply by abdonrd** [reply]
Ok! So this code would be fine, right?

`api/project/models/project.js`

```javascript
'use strict';

/**
 * Read the documentation (https://strapi.io/documentation/v3.x/concepts/models.html#lifecycle-hooks)
 * to customize this model
 */

const updateNumberOfProjects = async () => {
  const projectsCount = await strapi.query('project').count();
  // This page home is the single type. Note the `findOne`
  const pageHome = await strapi.query('page-home').findOne();

  await strapi
    .query('page-home')
    .update({ id: pageHome.id }, { numberOfProjects: projectsCount });
};

module.exports = {
  lifecycles: {
    async afterCreate(result, data) {
      await updateNumberOfProjects();
    },

    async afterDelete(result, params) {
      await updateNumberOfProjects();
    },
  },
};
```
